VCR recordings for ARM azure_mgmt_compute

vishrutshah opened the pull request:
Adding support for VCR recording in ARM azure_mgmt_compute service

This changes is part of the feature #291

azurecla commented:
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

veronicagg commented on resource_management/azure_mgmt_compute/spec/virtual_machines_spec.rb:
```diff
@@ -148,20 +150,12 @@
     # todo: add VM polling until VM is started if API method appear which provides
     # such info.
-    sleep 300
+    # sleep 300
```
should we condition this line for recording vs playback runs?

vishrutshah replied:
Yup that's better. Updated it based on one of the env variable that depends on recording vs playback.

veronicagg commented on resource_management/azure_mgmt_compute/spec/usage_operations_spec.rb:
```diff
@@ -8,8 +8,9 @@
-  before(:all) do
-    @client = COMPUTE_CLIENT.usage_operations
+  before(:each) do
+    @resource_helper = ResourceHelper.new
```
should we be instantiating everything that's in ResourceHelper every time? Don't we need only compute client in this case?

vishrutshah replied:
Yup that can be handled way better. updated it to use lazy initialization of class attributes.

veronicagg commented on resource_management/azure_mgmt_compute/spec/virtual_machines_spec.rb:
```diff
@@ -148,20 +150,12 @@
-
-  it 'should capture virtual machine' do
-    skip('no long running tasks should be performed') unless RUN_LONG_TASKS
+    #campuring VM requires VM to be generalized
```
typo in comment "capturing"

vishrutshah replied:
Corrected.

veronicagg commented:
@vishrutshah Left a few comments. What happens if someone runs arm:spec now? Also, should we update the readme with this change?

vishrutshah commented on resource_management/azure_mgmt_compute/spec/spec_helper.rb:
```diff
@@ -2,63 +2,84 @@
-  def to_b?
-    !self.to_i.zero?
+    @network_client = NetworkManagementClient.new(credentials)
+    @network_client.long_running_operation_retry_timeout = ENV['RETRY_TIMEOUT'].to_i || 30
```
`ENV['RETRY_TIMEOUT'].to_i || 30` will be always `0` in case of` ENV['RETRY_TIMEOUT']` is `nil`
Use `ENV.fetch('RETRY_TIMEOUT', 30).to_i`

vishrutshah commented:
arm:spec should have no effect on its use case. We have introduced arm:recorded to run the recorded tests. 

@veronicagg I've addressed the comments and refactored code with lazy initialization of class attributes. Please have a look once you get a chance. Thanks!

veronicagg commented:
@vishrutshah Updates look good to me. Thanks! As we chatted offline, with these changes arm:spec would run the long running operations by default, we could update arm:spec task to exclude running tests for the recorded gems, or wait to merge this PR until the other tests are already converted, so we can replace arm:spec task.

vishrutshah commented:
Sounds good!
